client/pages/home.jsx

```diff
@@ -1,31 +1,71 @@
 import React from 'react';
 import PostList from './post-list';
-
+import Modal from '../components/modal';
 export default class Home extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      clicked: false,
-      posts: []
+      postId: false,
+      posts: [],
+      open: false,
+      addedComment: false
     };
-    this.openEvent = this.openEvent.bind(this);
+    this.addComment = this.addComment.bind(this);
+    this.resetState = this.resetState.bind(this);
+    this.closeModal = this.closeModal.bind(this);
+  }
+
+  addComment(postId) {
+    this.setState({
+      postId: postId,
+      open: true
+    });
   }
 
-  openEvent() {
-    this.setState(prevState => ({
-      clicked: !prevState.clicked
-    }));
+  resetState() {
+    this.setState({
+      open: false,
+      addedComment: true
+    });
+  }
+
+  closeModal() {
+    this.setState({
+      open: false
+    });
   }
 
   componentDidMount() {
-    fetch('/api/posts/')
-      .then(response => response.json())
-      .then(posts => this.setState({ posts: posts }));
+    Promise.all([
+      fetch('/api/posts/'),
+      fetch('/api/comments/')
+    ])
+      .then(([res1, res2]) => Promise.all([res1.json(), res2.json()]))
+      .then(([data1, data2]) => this.setState({
+        posts: data1,
+        comments: data2
+      }));
+  }
+
+  componentDidUpdate() {
+    if (this.state.addedComment) {
+      fetch(`/api/comments/${this.state.comments.length + 1}`)
+        .then(res => res.json())
+        .then(data => this.setState({
+          comments: this.state.comments.concat(data)
+        }));
+      this.setState({
+        addedComment: false
+      });
```

Comment:
This isn't really what you want to do. Your Post should define a method for sending a new comment to the back end and then updating its own comments list.